DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/DllConflictManagment/DllConflictManager.cs

```diff
@@ -0,0 +1,245 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Autodesk.Revit.UI;
+
+namespace Speckle.Connectors.Revit.Plugin.DllConflictManagment;
+
+public class DllConflictManager
+{
+  private readonly Dictionary<string, AssemblyConflictInfo> _assemblyConflicts = new();
+  private readonly DllConflictManagmentOptionsLoader _optionsLoader;
+
+  public DllConflictManager(DllConflictManagmentOptionsLoader optionsLoader)
+  {
+    _optionsLoader = optionsLoader;
+  }
+
+  public void DetectConflicts()
+  {
+    Dictionary<string, Assembly> loadedAssembliesDict = new();
+    foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+    {
+      // TODO : what to do about multiple versions of loaded dlls?
+      loadedAssembliesDict[assembly.GetName().Name] = assembly;
+    }
+    LoadAssemblyAndDependencies(typeof(DllConflictManager).Assembly, loadedAssembliesDict, new HashSet<string>());
+  }
+
+  private void LoadAssemblyAndDependencies(
+    Assembly assembly,
+    Dictionary<string, Assembly> loadedAssemblies,
+    HashSet<string> visitedAssemblies
+  )
+  {
+    if (visitedAssemblies.Contains(assembly.GetName().Name))
+    {
+      return;
+    }
+    visitedAssemblies.Add(assembly.GetName().Name);
+
+    foreach (var assemblyName in assembly.GetReferencedAssemblies())
+    {
+      if (visitedAssemblies.Contains(assemblyName.Name))
+      {
+        continue;
+      }
+
+      if (loadedAssemblies.TryGetValue(assemblyName.Name, out Assembly? loadedAssembly))
+      {
+        if (loadedAssembly.GetName().Version != assemblyName.Version)
+        {
+          _assemblyConflicts[assemblyName.Name] = new(assemblyName, loadedAssembly);
+          continue; // if we already know there is a conflict here, no need to continue iterating dependencies
```

Comment:
Maybe. Nothing like hitting one error at a time though, maybe worth knowing ALL the problems we can see?